src/main/java/edu/wpi/first/pathweaver/Field.java

```diff
@@ -12,13 +13,38 @@
 
 
 public class Field {
+  private Supplier<Image> imageSupplier;
   private Image image;
   private Quantity<Length> rWidth;
   private Quantity<Length> rLength;
   private double scale;
   private Point2D coord;
   public Unit<Length> unit;
-  public Unit<Length> pixel;
+
+  /**
+   * Creates a new Field Object.
+   *
+   * @param imageSupplier a supplier for image of the field
```

Comment:
I don't think this imageSupplier is used anywhere

Reply:
It's used in `getImage()` to do a lazy initialization. It's to make tests work when there's no FX toolkit to do the low-level image loading